apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/dao/appdata/UnconfirmedTransactionTable.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c)  2018-2020. Apollo Foundation.
+ */
+
+package com.apollocurrency.aplwallet.apl.core.dao.appdata;
+
+import com.apollocurrency.aplwallet.apl.core.app.Transaction;
+import com.apollocurrency.aplwallet.apl.core.app.UnconfirmedTransaction;
+import com.apollocurrency.aplwallet.apl.core.db.DbKey;
+import com.apollocurrency.aplwallet.apl.core.db.LongKeyFactory;
+import com.apollocurrency.aplwallet.apl.core.db.derived.EntityDbTable;
+import com.apollocurrency.aplwallet.apl.core.transaction.TransactionType;
+import com.apollocurrency.aplwallet.apl.util.injectable.PropertiesHolder;
+import org.json.simple.JSONObject;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Singleton
+public class UnconfirmedTransactionTable extends EntityDbTable<UnconfirmedTransaction> {
+
+    private final LongKeyFactory<UnconfirmedTransaction> transactionKeyFactory;
+    private int maxUnconfirmedTransactions;
+    private final Map<DbKey, UnconfirmedTransaction> transactionCache = new HashMap<>();
+    private final PriorityQueue<UnconfirmedTransaction> waitingTransactions;
+    private final Map<TransactionType, Map<String, Integer>> unconfirmedDuplicates = new HashMap<>();
+    private final Map<Transaction, Transaction> txToBroadcastWhenConfirmed = new ConcurrentHashMap<>();
+    private final Set<Transaction> broadcastedTransactions = Collections.newSetFromMap(new ConcurrentHashMap<>());
```

Comment:
All these fields should be separated from UnconfirmedTransactionTable and moved to special service, which will process and maintain mempoll state

Reply:
I don't think it's good to keep UnconfirmedTransaction caches away from place where UnconfirmedTransaction are managed at low level.